easybuild/easyblocks/e/easybuildmeta.py

```diff
@@ -66,6 +66,28 @@ def __init__(self, *args, **kwargs):
             # consider setuptools first, in case it is listed as a sources
             self.easybuild_pkgs.insert(0, 'setuptools')
 
+        # opt-in to using pip for recent version of EasyBuild, if:
+        # - EasyBuild is being installed for Python >= 3.6;
+        # - pip is available, and recent enough (>= 21.0);
+        # - use_pip is not specified;
+        pyver = sys.version.split(' ')[0]
+        pip = which('pip')
+        self.log.info("Python version: %s - pip: %s", pyver, pip)
+        if sys.version_info >= (3, 6) and pip and self.cfg['use_pip'] is None:
+            # try to determine pip version, ignore any failures that occur while doing so;
+            # problems may occur due changes in environment ($PYTHONPATH, etc.)
+            pip_version = None
+            try:
+                pip_version = det_pip_version()
+                self.log.info("Found Python v%s + pip: %s", pyver, pip_version)
```

Comment:
Shouldn't we be checking major/minor version matching between `python` and `pip`?

Reply:
Good point, using `python -m pip` is indeed a better approach.
That should be tackled in the `PythonPackage` easyblock, as you're doing in #2807 :+1: